dags/xcom_gcs_ds_k8sExecutor.py

```diff
@@ -0,0 +1,220 @@
+from typing import Container
+from airflow.decorators import task, dag
+from airflow.providers.google.cloud.hooks.bigquery import BigQueryHook
+
+from datetime import datetime
+from kubernetes.client import models as k8s
+
+
+import pandas as pd
+import numpy as np
+from sklearn.model_selection import RepeatedStratifiedKFold
+from sklearn.model_selection import cross_val_score
+from lightgbm import LGBMClassifier
+
+
+docs = """
+By default, Airflow stores all return values in XCom. However, this can introduce complexity, as users then have to consider the size of data they are returning. Futhermore, since XComs are stored in the Airflow database by default, intermediary data is not easily accessible by external systems.
+By using an external XCom backend, users can easily push and pull all intermediary data generated in their DAG in GCS.
+
+Additionally using this with the Kubernetes Executor allows more fine grained controlled over the resources, images, and dependencies for each step on a per task basis.
+"""
+
+
+# Pod Config for the load_data task. This sets resource requests and limits for cpu and memory so that the Pod has enough memory to execute the task.
+load_config={ "pod_override": k8s.V1Pod(
+                metadata=k8s.V1ObjectMeta(labels={"purpose": "load_bq_data"}),
+                spec=k8s.V1PodSpec(
+                    containers=[
+                        k8s.V1Container(
+                            name="base",
+                            resources=k8s.V1ResourceRequirements(
+                                limits={
+                                    "cpu": 1,
+                                    "memory": "8Gi"
+                                },
+                                requests={
+                                    "cpu": 0.5,
+                                    "memory": "5Gi"
+                                }
+                            )
+                            )
+                        ]
+                    )
+                )
+            }
+
+
+# Pod Config for train and fit tasks. This sets resource requests and limits for cpu and memory so that the Pod perform the tasks more quickly from minutes to seconds.
```

Comment:
```suggestion
# Pod Config for train and fit tasks. This sets resource requests and limits for cpu and memory so that the Pod can perform the tasks more quickly. Reduces execution time from minutes to seconds compared to without the use of `KubernetesExecutor`.
```